pyramid/file.go

```diff
@@ -0,0 +1,48 @@
+package pyramid
+
+import "os"
+
+// File is pyramid wrapper for os.file that triggers pyramid hooks for file actions.
+type File struct {
+	fh     *os.File
+	access *evictionControl
+
+	rPath relativePath
+
+	close func(size int64) error
+	size  int64
+}
+
+func (f *File) Read(p []byte) (n int, err error) {
+	f.access.touch(f.rPath)
+	return f.fh.Read(p)
+}
+
+func (f *File) ReadAt(p []byte, off int64) (n int, err error) {
+	f.access.touch(f.rPath)
+	return f.fh.ReadAt(p, off)
+}
+
+func (f *File) Write(p []byte) (n int, err error) {
```

Comment:
pyramid isn't an immutable Read-Only FS. You can write files to it, there's an API which is similar to `os.Create` which returns a file handle you can write to. It will only become immutable when that file is closed. 
However you're right about being able to write to a file which is being read. I'll add a bool to return err in such cases.